app/models/news/item.rb

```diff
@@ -67,5 +75,17 @@ def for_target(target_name)
     def normalise_ids(ids)
       Array.wrap(ids).map(&:to_i).compact.uniq
     end
+
+    def generate_or_normalise_slug!
+      current_slug = slug.presence || title.presence
+      return unless current_slug
+
+      normalised_slug = normalise_slug(slug.presence || title)
+      self.slug = normalised_slug if slug != normalised_slug
```

Comment:
Isn't it better not to mutate existing state and just to ask for a slug?  It seems like the state mutation isn't actually  needed and generally prima-donna functions are a smell

```ruby
def slug
  @slug ||= begin
    current_slug = slug.presence || title
    
    if current_slug.present?
      slug.downcase.gsub(/\s+/, '-').gsub(/[^a-z0-9-]/, '').first(MAX_SLUG_LENGTH)
    end
  end
end
```